resources/assets/js/components/EventTitle.vue

```diff
@@ -0,0 +1,33 @@
+<template>
+  <component :is="component">
+    <!-- eslint-disable-next-line -->
+    {{ event.venue ? event.venue : event.location }}<b-badge v-if="event.online" variant="primary" pill class="nounderline">{{ translatedOnline }}</b-badge>
+  </component>
+</template>
+<script>
+export default {
+  props: {
+    idevents: {
+      type: Number,
+      required: true
+    },
+    component: {
+      type: String,
+      required: true
+    }
+  },
+  computed: {
+    event() {
+      return this.$store.getters['events/get'](this.idevents)
```

Comment:
General Vue question - is it preferable to pass in an id to a component, then get the associated entry from the store, rather than passing in the entity itself?  (or depends on the context?)

Reply:
Yes.  I no longer remember exactly why, but I started off passing objects around and shifted over time to passing ids around.  

I think there may be some issues with reactivity, or possibly just performance.  Either it doesn't kick in, or what happens is that a small change to the object results in the whole child component being re-rendered because the prop has changed, whereas with the id approach it would just trigger re-render of the relevant bits of the DOM inside the child component.

Discuss at retro if not clear.